distrobuilder/main.go

```diff
@@ -478,18 +478,137 @@ func getDefinition(fname string, options []string) (*shared.Definition, error) {
+	exit
+fi
+
+# Determine systemd version
+for path in /usr/lib/systemd/systemd /lib/systemd/systemd; do
+	[ -e "${path}" ] || continue
+	systemd_version="$("${path}" --version | head -n1 | cut -d' ' -f2)"
+	break
+done
+
+## Fixes
+
+# Fix capabilities
+(
+	if [ "${systemd_version}" -ge 244 ]; then
+		# Create top level dropin
+		dropin_dir="/run/systemd/system/service.d"
+		mkdir -p "${dropin_dir}"
+		echo "[Service]" > "${dropin_dir}/lxc.conf"
+		[ "${systemd_version}" -ge 247 ] && echo "ProtectProc=default" >> "${dropin_dir}/lxc.conf"
+		echo "ProtectControlGroups=no" >> "${dropin_dir}/lxc.conf"
+		echo "ProtectKernelTunables=no" >> "${dropin_dir}/lxc.conf"
+	else
+		# Create per-unit dropin for all service units
+		find /etc/systemd /run/systemd /usr/lib/systemd -name "*.service" -type f | sed -r 's#/usr/lib/systemd/##;s#/etc/systemd/##g;s#/run/systemd/##g' | while read service_file; do
+			dropin_dir="/run/systemd/${service_file}.d"
+			mkdir -p "${dropin_dir}"
+			echo "[Service]" > "${dropin_dir}/lxc.conf"
+			[ "${systemd_version}" -ge 247 ] && echo "ProtectProc=default" >> "${dropin_dir}/lxc.conf"
+			[ "${systemd_version}" -ge 232 ] && echo "ProtectControlGroups=no" >> "${dropin_dir}/lxc.conf"
+			[ "${systemd_version}" -ge 232 ] && echo "ProtectKernelTunables=no" >> "${dropin_dir}/lxc.conf"
+		done
+	fi
+)
+
+# Network fix for fedora/34/cloud
+(
+	. /etc/os-release
+
+	if ! which cloud-init || ! which NetworkManger || ! [ -e /sys/class/net/eth0 ] || ! is_container || [ "${ID}" != "fedora" ] || [ "${VERSION_ID}" != "34" ]; then
+		exit
+	fi
+
+	cat <<-EOF > /run/systemd/system/network-connection-activate.service
+[Unit]
+Description=Activate connection
+After=NetworkManager-wait-online.service
+
+[Service]
+ExecStart=-/usr/bin/nmcli c up "System eth0"
+Type=oneshot
+RemainAfterExit=true
+
+[Install]
+WantedBy=default.target
+EOF
+
+	mkdir -p /run/systemd/system/default.target.wants
+	ln -s /run/systemd/system/network-connection-activate.service /run/systemd/system/default.target.wants/network-connection-activate.service
+)
+
+# NetworkManager fix for various distributions
+(
+	if ! which NetworkManager || ! [ -f /bin/ip ] || ! [ -e /sys/class/net/eth0 ]; then
+		exit
+	fi
+
+	cat <<-EOF > /run/systemd/system/network-device-down.service
+[Unit]
+Description=Turn off network device
+Before=NetworkManager.service
+
+[Service]
+ExecStart=-/bin/ip link set eth0 down
+Type=oneshot
+RemainAfterExit=true
+
+[Install]
+WantedBy=default.target
+EOF
+
+	mkdir -p /run/systemd/system/default.target.wants
+	ln -s /run/systemd/system/network-device-down.service /run/systemd/system/default.target.wants/network-device-down.service
+)
+
+# Fix for systemd-networkd
+(
+	. /etc/os-release
+
+	[ "${ID}" = "altlinux" ] || exit
+
+	mkdir -p /run/systemd/system/systemd-networkd.service.d
+
+	if [ -e /run/systemd/system/systemd-networkd.service.d/lxc.conf ]; then
+		echo "BindReadOnlyPaths=/sys" >> /run/systemd/system/systemd-networkd.service.d/lxc.conf
```

Comment:
nitpick: this will keep appending on every `systemctl daemon-reload`, if you've got another `[Section]` in there, worse things will happen.
IMO this should just unconditionally overwrite the whole file.